Update and enhance Kedro VSCode extension documentation (#4538)
kedro-org/kedro: docs/source/visual_studio_code_extension/set_up_vscode.md

![](../meta/images/vscode_run.png)

## Setting a custom Kedro project path

Starting with Kedro VS Code extension version 0.3.0, you can now specify a custom path to your Kedro project. This is useful when:

- Your Kedro project is not at the root of your workspace
- You want to work with a Kedro project that is outside your current workspace

> - You have multiple Kedro projects and want to switch between them

Reviewer: 💯

> ### Ways to set a custom path for Kedro project:
> 
> #### Using the command palette

Reviewer: Maybe

```suggestion
### Set up a custom path using the command palette
```

(and same with the other H4 below)

- Use all Kedro extension features with the specified project

### Example directory structure

If your Kedro project is nested within other folders, setting a custom project path can help the extension locate it. For example:

```
root
│   file001.txt
│
└───folder1
│   │   file011.txt
│   │   file012.txt
│   │
│   └───kedroProject  <-- Set this path
│       │   pyproject.toml
│       │   README.md
│       │   ...
│
└───folder2
    │   file020.txt
    │   file021.txt
```

In this case, you would set the Kedro project path to the absolute path of the `kedroProject` directory, such as `/Users/username/root/folder1/kedroProject`.

### Switching between multiple projects

If you work with multiple Kedro projects, you can switch between them by updating the project path setting. The extension will automatically detect the change and reconfigure itself to work with the newly specified project.

### Troubleshooting

If the extension doesn't recognise your Kedro project after setting a custom path:

1. Ensure the path points to a valid Kedro project (containing `pyproject.toml` with Kedro dependencies)
2. Check that the path is an absolute path, not a relative one

> 3. Reload VS Code if the changes don’t take effect.

Reviewer 2: Will there be any quick way to check the extension has picked up the new project path ? Like some kind of logger or console output which can help users confirm of the change ?

Author: @ravi-kumar-pilla Yes, after path update, Kedro server restarts and you can see console output with updated project path `"kedroProjectPath": "/Users/<user>/QB/kedro-viz-3/kedro-viz/demo-project"`
